src/main/java/com/team4/testingsystem/services/TestsService.java

```diff
@@ -0,0 +1,94 @@
+package com.team4.testingsystem.services;
+
+import com.team4.testingsystem.entities.Test;
+import com.team4.testingsystem.enums.Status;
+import com.team4.testingsystem.exceptions.TestNotFoundException;
+import com.team4.testingsystem.exceptions.UserNotFoundException;
+import com.team4.testingsystem.repositories.TestsRepository;
+import com.team4.testingsystem.repositories.UsersRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDateTime;
+
+@Service
+public class TestsService {
+
+    private TestsRepository testsRepository;
+
+    private UsersRepository usersRepository;
+
+
+    @Autowired
+    public TestsService(TestsRepository testsRepository, UsersRepository usersRepository) {
+        this.testsRepository = testsRepository;
+        this.usersRepository = usersRepository;
+    }
+
+
+    public Iterable<Test> getAll() {
+        return testsRepository.findAll();
+    }
+
+    public Test getById(long id) {
+        try {
+            return testsRepository.findById(id).get();
+        } catch (Exception e) {
+            throw new TestNotFoundException();
+        }
+    }
+
+    public long createForUser(long userId) {
+
+       try {
+           Test test = Test.newBuilder()
+                   .setUser(usersRepository.findById(userId).get())
+                   .setCreatedAt(LocalDateTime.now())
+                   .setStatus(Status.NOT_STARTED)
+                   .build();
+
+           testsRepository.save(test);
+
+           long id = test.getId();
+           return id;
+       }
+       catch (Exception e){
+           throw new UserNotFoundException();
+       }
+    }
+
+    public void start(long id){
+
+        if (!testsRepository.existsById(id)){
+            throw new TestNotFoundException();
+        }
+
+        testsRepository.start(LocalDateTime.now(), id);
+    }
+
+    public void finish (long id, int evaluation){
+
+        if (!testsRepository.existsById(id)){
+            throw new TestNotFoundException();
+        }
+
+        testsRepository.finish(LocalDateTime.now(), evaluation, id);
+    }
+
+    public void updateEvaluation(long id, int newEvaluation) {
+
+        if (!testsRepository.existsById(id)){
+            throw new TestNotFoundException();
+        }
+
+        testsRepository.updateEvaluation(LocalDateTime.now(), newEvaluation, id);
+    }
+
+    public void removeById(long id) {
+        try {
+            testsRepository.deleteById(id);
+        } catch (Exception e) {
+            throw new TestNotFoundException();
+        }
```

Comment:
All this try-catch constructions are strange... I think solution like this can be better:
```
if (!testsRepository.existsById(id)) {
    throw new TestNotFoundException();
}
testsRepository.deleteById(id);
```

Reply:
@maxim12321 
Not a better construction.
1. You are making 1 redundant request to DB
2. Between exists check and deletion commands can happen anything. Maybe it's not so trivial for deletion. For example, on create operation, you will have smth like that
2.1. Check test is not created
2.2. Try to save a test in DB (don't forget about multithreading, several instances, etc)
2.3. Catch exception in case 2.2 will fail.

So, the question here is, why do we need the 2.1 commands, if we will still need 2.2 and 2.3?